### ISSUE TITLE
Hash mismatch with nix run 

### ISSUE THREAD

**Issue by mr3iscuit** [cucumber-sp/yandex-music-linux#128]
# Command
NIXPKGS_ALLOW_UNFREE=1 nix run github:NixOS/nixpkgs/nixos-unstable#yandex-music --impure

# Output
error: hash mismatch in fixed-output derivation '/nix/store/18rv31c5yz9qhjr33z26kk4a7sbch4n0-source.drv':
         specified: sha256-y+T2ckrnhrOiiPKBUlnvDb4FwrIfbaIXwVi16AoX/bQ=
            got:    sha256-CYb7J9XqQN3aiVaJzWwx9r62J5W3Su7GiaDC98/ubjU=
error: 1 dependencies of derivation '/nix/store/rz0j6jcypdl7wx6swdyxs9b3vl64mkyn-yandex-music-5.18.2.drv' failed to build

**Comment by ein-shved** [comment]
Already in progress. Wait for reviewer to approve https://github.com/NixOS/nixpkgs/pull/350288

**Comment by ein-shved** [comment]
Please, check

**Comment by mr3iscuit** [comment]
Same problem

```nix
NIXPKGS_ALLOW_UNFREE=1 nix run github:NixOS/nixpkgs/nixos-unstable#yandex-music --impure
error: hash mismatch in fixed-output derivation '/nix/store/412czsdrkaqw9w4y5pi61vd7n1fph9kw-source.drv':
         specified: sha256-nhy4D2PgTMsQOs8hPY39Z+I+Tldgf1ASZbatfxWqNTw=
            got:    sha256-ch9pqn6BrO54mCdExluv2lGkmu5q6Vy2/OcNJKbslTo=
error: 1 dependencies of derivation '/nix/store/lvyayfcy7jpw4kmhnwlxc8cxw5yd54k1-yandex-music-5.23.2.drv' failed to build
```

**Comment by mr3iscuit** [comment]
In nix, the file to be used must not change if it is used in a nix derivation, and must be the same for each version of the nix derivation. The problem may be hidden on the source URL which the file (yandex music) is downloaded.

**Comment by cucumber-sp** [comment]
@ein-shved если я правильно понял он берет хеш для новой версии но ссылается на старый файл? Может используете теги с репозитория так же как я сделал для Arch linux

**Comment by ein-shved** [comment]
Нет, скачиваемый файл на стороне яндекса меняет иногда свой хэш. Тут ничего не поделать, кроме как жаловаться в сам яндекс.

**Comment by mr3iscuit** [comment]
А можно распаковать из репо другово дитрибутива ? например с arch ? но елси он не использует тот же самый скрипт который делаеть фетч из самого яндекса, потому что в таком случае хеш тоже меняется. Спасибо за внимание.
